### PR TITLE
CLI documentation

### PR DESCRIPTION
**Pull request by ernestum**
## Description

Add some documentation for the command line interface.

## Testing

Documentation only changes.

### CODE REVIEW

**Comment by RedTachyon** [comment]
One thing worth mentioning (discussed yesterday with Adam) is that if an environment is registered upon importing a package, or in a certain file (e.g. inside `custom_env.py` you call `gym.register("CustomEnv-v0")`), then in the CLI interface you can train with that environment by providing the package/file name in the env ID (e.g. `custom_env:CustomEnv-v0`)

Some information about this functionality is in the [Gymnasium docs](https://gymnasium.farama.org/api/registry/). The same thing worked in gym==0.21.0 as well

File: docs/getting-started/cli.rst [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+.. code-block:: bash
+
+    python -m imitation.scripts.train_adversarial airl with \
+        seals_mountain_car \
+        total_timesteps=5000 \
+        expert.policy_type=ppo-huggingface \
+        demonstrations.n_expert_demos=500
+
+Note: the small number of total timesteps is only for demonstration purposes and will not produce a good policy.
```
Comment (levmckinney) [inline docs/getting-started/cli.rst:77]: Use the `.. note:` environment.

File: docs/getting-started/cli.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+
+Run GAIL on the `seals/Swimmer-v0` environment
```
Comment (levmckinney) [inline docs/getting-started/cli.rst:80]: Use double back ticks ` ``this`` ` for literals.

File: src/imitation/scripts/ingredients/expert.py [hunk 3]
```diff
@@ -1,4 +1,19 @@
-"""Common configuration elements for loading of expert policies."""
+"""This ingredient provides an expert policy.
+
+The expert policy is either loaded from disk or from the HuggingFace Model Hub or is
```
Comment (levmckinney) [inline src/imitation/scripts/ingredients/expert.py:3]: This is so much better thank you!

**Review by levmckinney — COMMENTED** [review]
Looks really good. A couple of minor suggestions. Ping me for a rereview when you're ready.

File: docs/getting-started/cli.rst [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+.. code-block:: bash
+
+    python -m imitation.scripts.<script> print_config
+
+Run BC on the `CartPole-v1` environment with a pre-trained PPO policy as expert:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Note: the cartpole environment is specified via a named configuration
+
+.. code-block:: bash
```
[the remark this answers is not in the document]
Comment (ernestum) [reply docs/getting-started/cli.rst:40]: You are right with this concern!

Unfortunately this is not so easy to do. See https://stackoverflow.com/questions/71497528/can-i-modify-doctest-to-test-bash-inside-sphinx-doc

We would either have to use python code and run the commands via subprocess or use some alpha-stage python module also does not quite exactly do what we need (we would have to re-write the documentation in Markdown). 

Given that the CLI is going to change soon anyway, I would not put in the effort to test it right now. What do you think?

[the remark this answers is not in the document]
Comment (ernestum) [reply docs/getting-started/cli.rst:40]: The python module in question: https://github.com/jamesbehr/docshtest

[the remark this answers is not in the document]
Comment (levmckinney) [reply docs/getting-started/cli.rst:40]: Sounds good lets just make sure to add a task to the issue tracker for the CI overall to update this part of the documentation.

**Review by levmckinney — APPROVED** [review]
LGTM

File: docs/getting-started/cli.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (ernestum) [reply docs/getting-started/cli.rst:40]: I added it to the bottom of the description here: https://github.com/HumanCompatibleAI/imitation/pull/703

**Pull request merged**